cpg_workflows/jobs/joint_genotyping.py

```diff
@@ -166,6 +191,36 @@ def make_joint_genotyping_jobs(
         if siteonly_j:
             jobs.append(siteonly_j)
 
+        # split multi-allelics for all those VCF fragments
+        # do this before gathering into one huge VCF
+        split_multi_job, split_fragment = add_split_multiallelics_job(
+            b=b,
+            input_vcf=unsplit_vcfs[idx],
+            output_vcf_path=split_jc_vcf_path,
+            job_attrs=(job_attrs or {}) | dict(part=f'{idx + 1}/{scatter_count}'),
+        )
+
+        # make a job to split this fragment's multiallelic variantt
+        vcfs.append(split_fragment)
+        if split_multi_job:
+            split_multi_job.depends_on(jobs[-1])
+            jobs.append(split_multi_job)
+
+        # if requested, make a site-only VCF for this split fragment (for VEP)
+        if out_split_sitesonly_vcf_part_paths:
+            split_sitesonly_jc_vcf_path = out_split_sitesonly_vcf_part_paths[idx]
+            siteonly_j, siteonly_j_vcf = add_make_sitesonly_job(
+                b=b,
+                input_vcf=split_fragment,
+                output_vcf_path=out_split_sitesonly_vcf_part_paths[idx],
+                job_attrs=(job_attrs or {}) | dict(part=f'{idx + 1}/{scatter_count}'),
+            )
+
+            # depend on the previous job
+            if siteonly_j:
+                siteonly_j.depends_on(jobs[-1])
```

Comment:
nit: I am not sure how I feel about the use of jobs[-1] in this context.

Reply:
It's safer to depend on all prior jobs, and Hail can do much bigger dependency graphs than this. Will update